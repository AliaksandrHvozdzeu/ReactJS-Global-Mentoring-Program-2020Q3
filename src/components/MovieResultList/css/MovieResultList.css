.movie-result {
  color: var(--white);
}

.movie-result-title {
  height: 5rem;
  font-size: 1.5rem;
  display: flex;
  align-items: center;
}

.movie-result-count::after {
  content: '';
  display: inline-block;
  width: 0.5rem;
}

.movie-result-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  column-gap: 3rem;
  row-gap: 3rem;
  justify-items: start;
  margin: 1rem auto 2rem;
}
