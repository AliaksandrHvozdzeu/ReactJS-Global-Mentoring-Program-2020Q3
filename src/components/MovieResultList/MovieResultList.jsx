import React from 'react';
import PropTypes from 'prop-types';
import Movie from '../Movie';
import './style.css';

const SearchResultCount = ({ count }) => (
  <div className="movie-result-title">
    <p className="movie-result-text">
      <span className="movie-result-count">{count}</span>
      movies found
    </p>
  </div>
);

SearchResultCount.propTypes = {
  count: PropTypes.number.isRequired,
};

const MovieResultList = ({ results }) => (
  <section className="movie-result">
    <SearchResultCount count={results.length} />
    <div className="movie-result-list">
      {results.map((movieDetails) => (
        <Movie key={movieDetails.id} details={movieDetails} />
      ))}
    </div>
  </section>
);

MovieResultList.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      poster: PropTypes.string.isRequired,
      genres: PropTypes.array.isRequired,
      release: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired
    }).isRequired
  ).isRequired
};

export default MovieResultList;
