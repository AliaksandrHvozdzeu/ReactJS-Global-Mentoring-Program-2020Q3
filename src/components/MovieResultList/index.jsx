import React from 'react';
import PropTypes from 'prop-types';
import Movie from '../Movie';
import './style.css';

const SearchResultCount = ({ count }) => (
  <div className="movie-result-title">
    <p className="movie-result-text">
      <span className="movie-result-count">{count}</span>
      movies found
    </p>
  </div>
);

SearchResultCount.propTypes = {
  count: PropTypes.number.isRequired,
};

const MovieResultList = ({ results }) => (
  <section className="movie-result">
    <SearchResultCount count={results.length} />
    <div className="movie-result-list">
      {results.map((movieDetails) => (
        <Movie key={movieDetails.id} details={movieDetails} />
      ))}
    </div>
  </section>
);

MovieResultList.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      genre: PropTypes.string.isRequired,
      release: PropTypes.string.isRequired,
      rating: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};

export default MovieResultList;
