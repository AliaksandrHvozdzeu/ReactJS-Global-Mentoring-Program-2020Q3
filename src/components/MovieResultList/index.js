import MovieResultList from './MovieResultList';

export default MovieResultList;
