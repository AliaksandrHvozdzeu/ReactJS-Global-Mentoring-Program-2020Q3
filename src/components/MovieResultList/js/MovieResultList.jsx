import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Movie from "../../Movie";
import SearchResultCount from "./SearchResultCount";
import NotFoundMovie from "../../NotFoundMovie";
import "../css/MovieResultList.css";

const MovieResultList = ({ movies }) => {

  const isMoviesNotFound = () => {
    if (movies.length === 0) {
      return (
        <section className="movie-result">
          <NotFoundMovie />
        </section>
      );
    }
    return (
      <>
        <section className="movie-result">
          <SearchResultCount count={movies.length} />
          <div className="movie-result-list">
            {movies.map((movieDetails) => (
              <Movie key={`${movieDetails.id}`} details={movieDetails} />
            ))}
          </div>
        </section>
      </>
    );
  };

  return (
    <>
      {isMoviesNotFound()}
    </>
  );

};

MovieResultList.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      overview: PropTypes.string.isRequired,
      poster_path: PropTypes.string.isRequired,
      genres: PropTypes.arrayOf(PropTypes.string),
      release_date: PropTypes.string.isRequired,
      vote_average: PropTypes.number
    }).isRequired
  ).isRequired
};

const mapStateToProps = (state) => ({ movies: state.movies.movies });

export default connect(mapStateToProps, null)(MovieResultList);
