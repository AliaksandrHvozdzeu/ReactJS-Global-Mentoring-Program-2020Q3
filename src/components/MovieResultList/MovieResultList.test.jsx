import React from "react";
import rerender from "react-test-renderer";
import MovieResultList from "./js/MovieResultList";
import configureStore from "redux-mock-store";
import { jest } from "@jest/globals";
import { Provider } from "react-redux";

describe("when SearchResultCount", () => {

  const movies = [
      {
        "id": 181808,
        "title": "Star Wars: The Last Jedi",
        "tagline": "The Saga Continues",
        "vote_average": 7.1,
        "vote_count": 4732,
        "release_date": "2017-12-13",
        "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
        "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
        "budget": 200000000,
        "revenue": 1325937250,
        "genres": [
          "Fantasy",
          "Adventure",
          "Science Fiction"
        ],
        "runtime": 152
      },
      {
        "id": 1892,
        "title": "Return of the Jedi",
        "tagline": "The Empire Falls...",
        "vote_average": 7.9,
        "vote_count": 6037,
        "release_date": "1983-05-23",
        "poster_path": "https://image.tmdb.org/t/p/w500/jx5p0aHlbPXqe3AH9G15NvmWaqQ.jpg",
        "overview": "As Rebel leaders map their strategy for an all-out attack on the Emperor's newer, bigger Death Star. Han Solo remains frozen in the cavernous desert fortress of Jabba the Hutt, the most loathsome outlaw in the universe, who is also keeping Princess Leia as a slave girl. Now a master of the Force, Luke Skywalker rescues his friends, but he cannot become a true Jedi Knight until he wages his own crucial battle against Darth Vader, who has sworn to win Luke over to the dark side of the Force.",
        "budget": 32350000,
        "revenue": 572700000,
        "genres": [
          "Adventure",
          "Action",
          "Science Fiction"
        ],
        "runtime": 135
      }
    ];

  const initialStore = {
    "movies": {
      "movies": [
        {
          "id": 181808,
          "title": "Star Wars: The Last Jedi",
          "tagline": "The Saga Continues",
          "vote_average": 7.1,
          "vote_count": 4732,
          "release_date": "2017-12-13",
          "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
          "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
          "budget": 200000000,
          "revenue": 1325937250,
          "genres": [
            "Fantasy",
            "Adventure",
            "Science Fiction"
          ],
          "runtime": 152
        },
        {
          "id": 1892,
          "title": "Return of the Jedi",
          "tagline": "The Empire Falls...",
          "vote_average": 7.9,
          "vote_count": 6037,
          "release_date": "1983-05-23",
          "poster_path": "https://image.tmdb.org/t/p/w500/jx5p0aHlbPXqe3AH9G15NvmWaqQ.jpg",
          "overview": "As Rebel leaders map their strategy for an all-out attack on the Emperor's newer, bigger Death Star. Han Solo remains frozen in the cavernous desert fortress of Jabba the Hutt, the most loathsome outlaw in the universe, who is also keeping Princess Leia as a slave girl. Now a master of the Force, Luke Skywalker rescues his friends, but he cannot become a true Jedi Knight until he wages his own crucial battle against Darth Vader, who has sworn to win Luke over to the dark side of the Force.",
          "budget": 32350000,
          "revenue": 572700000,
          "genres": [
            "Adventure",
            "Action",
            "Science Fiction"
          ],
          "runtime": 135
        }
      ]
    }
  };

  const initialStoreEmptyList = {
    "movies": {
      "movies": []
    }
  };

  test("then snapshot created", () => {
    const store = configureStore([])(initialStore);
    store.dispatch = jest.fn();
    const component = rerender.create(<Provider store={store}>
      <MovieResultList movies={movies} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("then snapshot created when movies list is empty", () => {
    const store = configureStore([])(initialStoreEmptyList);
    store.dispatch = jest.fn();
    const component = rerender.create(<Provider store={store}>
      <MovieResultList movies={[]} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
