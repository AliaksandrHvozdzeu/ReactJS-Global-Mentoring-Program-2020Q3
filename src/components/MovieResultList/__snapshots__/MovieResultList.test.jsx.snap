// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when SearchResultCount then snapshot created 1`] = `
<section
  className="movie-result"
>
  <div
    className="movie-result-title"
  >
    <p
      className="movie-result-text"
    >
      <span
        className="movie-result-count"
      >
        2
      </span>
      movies found
    </p>
  </div>
  <div
    className="movie-result-list"
  >
    <div
      className="movie-result-item movie"
    >
      <img
        alt="movie poster"
        className="movie-logo"
        onClick={[Function]}
        onError={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg"
      />
      <div
        className="movie-description"
      >
        <button
          className="movie-title"
          onClick={[Function]}
          type="button"
        >
          Star Wars: The Last Jedi
        </button>
        <p
          className="movie-genre"
        >
          Fantasy, Adventure, Science Fi...
        </p>
        <p
          className="movie-release"
        >
          2017
        </p>
      </div>
      <button
        className="movie-menu movie-menu-closed"
        onClick={[Function]}
        type="button"
      >
        <div
          className="movie-menu-button"
        />
      </button>
    </div>
    <div
      className="movie-result-item movie"
    >
      <img
        alt="movie poster"
        className="movie-logo"
        onClick={[Function]}
        onError={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w500/jx5p0aHlbPXqe3AH9G15NvmWaqQ.jpg"
      />
      <div
        className="movie-description"
      >
        <button
          className="movie-title"
          onClick={[Function]}
          type="button"
        >
          Return of the Jedi
        </button>
        <p
          className="movie-genre"
        >
          Adventure, Action, Science Fic...
        </p>
        <p
          className="movie-release"
        >
          1983
        </p>
      </div>
      <button
        className="movie-menu movie-menu-closed"
        onClick={[Function]}
        type="button"
      >
        <div
          className="movie-menu-button"
        />
      </button>
    </div>
  </div>
</section>
`;

exports[`when SearchResultCount then snapshot created when movies list is empty 1`] = `
<section
  className="movie-result"
>
  <section
    className="movie-no-results"
  >
    No Movie Found
  </section>
</section>
`;
